### PR TITLE
Issue #15760: Updated JavadocParagraph examples

### LINKED ISSUE #15760: JavadocParagraph documentation example s should be updated to same by structure
Detected at https://github.com/checkstyle/checkstyle/pull/15746

Rules on how examples should be described at https://github.com/checkstyle/checkstyle/issues/13345

### PR DESCRIPTION
**Pull request by akankshat05**
Closes #15760 and #15689

### CODE REVIEW

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java [hunk 1]
```diff
@@ -16,12 +16,38 @@
  * <p>No blank line before the tag (violation).
  *
  * <p>
- * New line after tag (violation).
+ * New line after tag (ok).
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:19]: Lets remove all ok and violation words.
If no violation it is ok.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java [hunk 2]
```diff
@@ -16,12 +16,38 @@
...
+   * </p>
+   */
+  void foo2() {}
+  /**
+   * Double newline.
+   *
+   * Some Paragraph.
+   */
+  // violation 3 lines above 'Empty line should be followed by <p> tag'
+  void foo3() {}
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:51]: Placing `// ...` between javadoc and method signature is a bit weird style.
Lets try to move this comment above javadoc.

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:51]: Please apply the same to all other cases like this.

**Review by romani — CHANGES_REQUESTED** [review]
Very good.
Items to make it awesome:

**Comment by Zopsss** [comment]
Github, generate site

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java [hunk 3]
```diff
@@ -10,18 +10,43 @@
...
  */
 // violation 3 lines above 'tag should be placed immediately before the first word'
+
 public class Example1 {
+  // violation 4 lines below '<p> tag should not precede HTML block-tag '<pre>''
+  /**
+   * No tag
+   *
+   * <p>
+   * <pre>item 1</pre>
```
Comment (Zopsss) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:32]: Can you also add an example of a inline tag preceded by `<p>` tag?

```
<p><b>this is okay, B tag is an inline tag</b></p>
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java [hunk 4]
```diff
@@ -1,53 +0,0 @@
...
-   * <tbody>
-   * <p>
-   * <tr>
-   * nested paragraph preceding block tag, this allowed.
-   * </tr>
-   * </tbody>
-   * </table>
-   *
-   */
-  void foo2() {}
```
Comment (Zopsss) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java:51]: please add this example back, it is an important example for user to understand Check's working.

Comment (akankshat05) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java:51]: I have removed the example, as the maximum file lines could be 60 only, and all the violation messages were covered in Example1 and Example2.

Comment (Zopsss) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java:51]: @romani should we add suppression for this? or is it ok to exclude this example?

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java:51]: we should have all message examples, if some are duplicated, we do not need them.
if you think that removed showing something special we can keep it. but we might need to explain user what is so special in such case.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java (hunk 3 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:32]: @akankshat05, is this done?

Please respond done to each review items it helps a lot.

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java [hunk 5]
```diff
@@ -10,18 +10,43 @@
...
  *
- * <p> Whitespace after tag (violation).
+ * <p> Whitespace after tag
  *
+ * <p><b>p tag before inline tag B, this is ok</b></p>
  */
-// violation 3 lines above 'tag should be placed immediately before the first word'
+// violation 4 lines above 'tag should be placed immediately before the first word'
 public class Example1 {
+  // violation 4 lines below '<p> tag should not precede HTML block-tag '<pre>''
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:27]: Please make Examples to be same by size, Linux numbers.
Please position  classes and methods to same line numbers.
You can archive this by placing empty line before class

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:27]: Put few empty lines after class to keep foo1 on same line number in both examples.

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:27]: Put 2 empty line after foo1 to keep foo2 to be on same line numbers

**Review by romani — CHANGES_REQUESTED** [review]
Please pay attention to example images in issue description 

Items:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java (hunk 4 above)
Comment (akankshat05) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example3.java:51]: Okay, I will update the examples to include the Example3 case by adding the suppression.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java (hunk 5 above)
Comment (akankshat05) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:27]: Sure, doing it

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java (hunk 3 above)
Comment (akankshat05) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java:32]: This is done.

**Comment by akankshat05** [comment]
@romani @Zopsss I have updated the PR with the requested changes. Please review them. Thank you.

**Comment by Zopsss** [comment]
Github, generate website

**Comment by Zopsss** [comment]
Job pr-age CI is failing. Please rebase your branch with master

**Comment by akankshat05** [comment]
> Job pr-age CI is failing. Please rebase your branch with master

Ok doing it

**Comment by romani** [comment]
GitHub, generate website

**Review by Zopsss — APPROVED** [review]
LGTM. Thanks for adding back the example 

@romani the first example has extra whitespaces, is it okay to have such spaces?

**Comment by romani** [comment]
https://www.diffchecker.com/K1thTTCw/

Java code should be in same line numbers.

**Comment by romani** [comment]
Extra empty lines are required sometimes to let java code to stay on same line numbers. It helps a lot user to spot a difference visually.

**Comment by romani** [comment]
following extra empty lines were removed to make Examples be same in structure and size:
```
✔ ~/java/github/romani/checkstyle [akankshat05/javadocparagraph L|✚ 1] 
 $ git diff
diff --git a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java
index be3dc68..236f69b 100644
--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/Example1.java
@@ -24,8 +24,6 @@ package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocparagraph;
  */
 // violation 4 lines above 'tag should be placed immediately before the first word'
 
-
-
 public class Example1 {
 
 
✔
```

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
Github, generate site

**Comment by romani** [comment]
https://www.diffchecker.com/QguOv6zT/

**Comment by romani** [comment]
@akankshat05 , please pay attention to extra changes I did

**Comment by akankshat05** [comment]
> @akankshat05 , please pay attention to extra changes I did

Okay, sure. Thank you.

**Pull request merged**